test_bst.py

```diff
@@ -0,0 +1,58 @@
+#! /usr/bin/env python
+
+import pytest
+from bst import Bst
+
+
+def test_constructor():
+    instance = Bst()
+    assert type(instance) is Bst
+
+
+def test_insert(empty_tree):
+    a = empty_tree
+    a.insert(5)
+    a.insert(7)
+    assert 7 is a.top.right.data
+    a.insert(8)
+    a.insert(6)
+    a.insert(9)
+    assert a.top.data == 5
+    a.insert(9)
+    #test that a duplicate value does not get added
+    assert a.size() == 5
```

Comment:
You should probably have an assertion about the size of `a` before you re-check to ensure that the value didn't change when you inserted a duplicate value.